javascript/javascript1/week3/homework.js

```diff
@@ -0,0 +1,199 @@
+}
+logOutSeriesText(seriesDurations);
+
+// Smart-ease - Back to the basics!
+// NOnoN0nOYes (Note taking app)
+
+const notes = [];
+
+// Save a note
+
+function saveNote(content, id) {
+    const newObjects = { content, id };
+    if (content === "" || typeof id !== "number") {
+        console.log("Please enter information as requested");
+    }
+    else {
+        notes.push(newObjects);
+    }
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+// Get a note
+
+function getNote(id) {
+    for (var i = 0; i < notes.length; i++) {
+        if (id === notes[i].id) {
+            return notes[i];
+        }
+    }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+// Log out notes
+
+function logOutNotesFormatted() {
+    for (var j = 0; j < notes.length; j++) {
+        console.log(`The note with id: ${notes[j].id}, has the following note text: ${notes[j].content}`);
+    }
+}
+
+logOutNotesFormatted(); // should log out the text below
+
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+
+// CactusIO-interactive (Smart phone usage app)
+
+//Adding an activity
+
+const activities = [];
+var today = new Date().toLocaleDateString("en-US");
+
+function addActivity(date, activity, duration) {
+    const activityObject = { date, activity, duration };
+    activities.push(activityObject);
+}
+
+addActivity("23/7-18", "Youtube", 10);
+addActivity(today, "Youtube", 30);
+addActivity(today, "Facebook", 50);
+console.log(activities);
+
+/*
+activities should now look like this
+[{
+    date: '23/7-18',
+    activity: 'Youtube',
+    duration: 30,
+}]
+*/
+
+// Show my status(Usage limit and extra features)
+
+function showStatus(activities) {
+    let sum = 0;
+    let limit = 100;
+    let todaySum = 0;
+    let todayActivity = 0;
+    for (var i = 0; i < activities.length; i++) {
+        sum += activities[i].duration;
+        if (activities[i].date === today) {
+            todaySum += activities[i].duration;
+            todayActivity++;
+        }
+    }
+    if (activities.length === 0) {
+        console.log("Add some activities before calling showStatus");
+    }
+    else if (sum > limit) {
+        console.log("You have reached your limit, no more smartphoning for you!");
+    }
+    else {
+        console.log(`You have added ${i} activities. They amount to ${sum} min. of usage`);
+        console.log(`You have added ${todayActivity} activities today. They amount to ${todaySum} min. of usage`)
+    }
+}
+
+showStatus(activities); // will log out this "You have added 3 activities. They amount to 78 min. of usage"
+
+function findMaxDuration(activities) {
+    const durationArray = [];
+    for (var j = 0; j < activities.length; j++) {
```

Comment:
Unless you need `j` outside the loop, use `let j = 0` so that `j` will only exist inside the loop and will not conflict with anything outside the loop.